src/components/MapComponent.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useRef, useState, useCallback } from 'react';
+import { GoogleMap, LoadScript } from '@react-google-maps/api';
+
+const containerStyle = {
+  width: '100%',
+  height: '400px',
+};
+
+interface MapComponentProps {
+  onLocationUpdate: (latitude: number, longitude: number) => void;
+}
+
+const MapComponent: React.FC<MapComponentProps> = ({ onLocationUpdate }) => {
+  const [latitude, setLatitude] = useState<number | null>(null);
+  const [longitude, setLongitude] = useState<number | null>(null);
+  const [callCount, setCallCount] = useState<number>(0);
+
+  // eslint-disable-next-line no-undef
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+
+  // Control flag to limit API calls in development
+  const isDevelopment = process.env.NODE_ENV === 'development';
+
+  //const apiKey = process.env.REACT_APP_GOOGLE_MAPS_API_KEY; // this was my inital approch but its not working as expected
```

Comment:
when you push this to main make sure you remove the comment that has the API Key and leave the string empty.

If you like you can try and vaulting the API KEY in this branch. If not then we can go ahead and make a new story for that.

What I would recommend is put it in a seperate json config file and be sure to put that file under `.gitignore`. This config file will have all secret variables we need and then you can just import the variable into where you need it.

Reply:
Here are other [approaches](https://www.darkreading.com/cybersecurity-operations/how-do-i-protect-my-api-keys-from-appearing-in-github-search-results-). 

Env variables are another options but would make local setup more difficult since it involves messing with system settings, especially for mac users (sorry moe).

Reply:
API key removed and story created